Thank you for the example — I see zero diagnostics returned from eslint for that, which is concerning.
![image](https://github.com/zed-industries/zed/assets/2690773/ab50cee7-2786-48c4-ac37-e32a27a315e0)

What also is odd for me, that `npm it` (and `npm run test`) are both erroring for me?

```
Last login: Wed Feb 21 14:46:42 on ttys000
npm ru%
~/Downloads/eslint-configs-demo main:origin/main ❯ npm run test

> eslint-configs-demo@1.0.0 test
> eslint index.js


Oops! Something went wrong! :(

ESLint: 8.56.0

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/someonetoignore/Downloads/eslint-configs-demo/.eslintrc.js from /Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs not supported.
.eslintrc.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename .eslintrc.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/someonetoignore/Downloads/eslint-configs-demo/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

    at module.exports [as default] (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/import-fresh/index.js:32:59)
    at loadJSConfigFile (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2583:47)
    at loadConfigFile (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2667:20)
    at ConfigArrayFactory.loadInDirectory (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2877:34)
    at CascadingConfigArrayFactory._loadConfigInAncestors (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3871:46)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3792:18)
    at FileEnumerator._iterateFilesWithFile (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli-engine/file-enumerator.js:368:43)
    at FileEnumerator._iterateFiles (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli-engine/file-enumerator.js:349:25)
    at FileEnumerator.iterateFiles (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli-engine/file-enumerator.js:299:59)
    at iterateFiles.next (<anonymous>)
    at CLIEngine.executeOnFiles (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli-engine/cli-engine.js:797:48)
    at ESLint.lintFiles (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/eslint/eslint.js:551:23)
    at Object.execute (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli.js:421:36)
    at async main (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/bin/eslint.js:152:22)

~/Downloads/eslint-configs-demo main:origin/main ❯ npm it

up to date, audited 100 packages in 442ms

23 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> eslint-configs-demo@1.0.0 test
> eslint index.js


Oops! Something went wrong! :(

ESLint: 8.56.0

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/someonetoignore/Downloads/eslint-configs-demo/.eslintrc.js from /Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs not supported.
.eslintrc.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename .eslintrc.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/someonetoignore/Downloads/eslint-configs-demo/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

    at module.exports [as default] (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/import-fresh/index.js:32:59)
    at loadJSConfigFile (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2583:47)
    at loadConfigFile (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2667:20)
    at ConfigArrayFactory.loadInDirectory (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:2877:34)
    at CascadingConfigArrayFactory._loadConfigInAncestors (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3871:46)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/@eslint/eslintrc/dist/eslintrc.cjs:3792:18)
    at FileEnumerator._iterateFilesWithFile (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli-engine/file-enumerator.js:368:43)
    at FileEnumerator._iterateFiles (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli-engine/file-enumerator.js:349:25)
    at FileEnumerator.iterateFiles (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli-engine/file-enumerator.js:299:59)
    at iterateFiles.next (<anonymous>)
    at CLIEngine.executeOnFiles (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli-engine/cli-engine.js:797:48)
    at ESLint.lintFiles (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/eslint/eslint.js:551:23)
    at Object.execute (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/lib/cli.js:421:36)
    at async main (/Users/someonetoignore/Downloads/eslint-configs-demo/node_modules/eslint/bin/eslint.js:152:22)
npm ERR! code 2
npm ERR! path /Users/someonetoignore/Downloads/eslint-configs-demo
npm ERR! command failed
npm ERR! command sh -c eslint index.js

npm ERR! A complete log of this run can be found in: /Users/someonetoignore/.npm/_logs/2024-02-21T12_55_23_599Z-debug-0.log

~/Downloads/eslint-configs-demo main:origin/main ❯
```

so I cannot really say that it works for me, same thing is in VSCode:
![image](https://github.com/zed-industries/zed/assets/2690773/276cda75-1921-4654-be7e-d073138f0b80)

I have also tried running that `pnpm` and converting the config module to something else to no avail, and not sure how to proceed with my poor frontend knowledge.
It something off with my development set-up there?

I would love to get down to fixing eslint there in VSCode and Zed first, and then test this PR before merging it.

--- a/crates/zed/src/languages/typescript.rs
+++ b/crates/zed/src/languages/typescript.rs
@@ -237,6 +237,9 @@ impl LspAdapter for EsLintLspAdapter {
                     "name": workspace_root.file_name()
                         .unwrap_or_else(|| workspace_root.as_os_str()),
                 },
+                "experimental": {
+                    "useFlatConfig": workspace_root.join("eslint.config.js").exists()
+                }
             }
         })
     }

--- a/crates/zed/src/languages/typescript.rs
+++ b/crates/zed/src/languages/typescript.rs
@@ -238,8 +238,8 @@ impl LspAdapter for EsLintLspAdapter {
                         .unwrap_or_else(|| workspace_root.as_os_str()),
                 },
                 "experimental": {
-                    "useFlatConfig": workspace_root.join("eslint.config.js").exists()
-                }
+                    "useFlatConfig": workspace_root.join("eslint.config.js").exists(),
+                },
             }
         })
     }

--- a/crates/zed/src/languages/typescript.rs
+++ b/crates/zed/src/languages/typescript.rs
@@ -238,7 +238,7 @@ impl LspAdapter for EsLintLspAdapter {
                         .unwrap_or_else(|| workspace_root.as_os_str()),
                 },
                 "experimental": {
-                    "useFlatConfig": workspace_root.join("eslint.config.js").exists(),
+                    "useFlatConfig": workspace_root.join("eslint.config.js").is_file(),
                 },
             }
         })

